src/Component/src/Metadata/Resource/Factory/MutatorResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Resource\Factory;
+
+use Sylius\Resource\Metadata\Mutator\OperationMutatorCollectionInterface;
+use Sylius\Resource\Metadata\Mutator\ResourceMutatorCollectionInterface;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\Metadata\Operations;
+use Sylius\Resource\Metadata\Resource\ResourceMetadataCollection;
+use Sylius\Resource\Metadata\ResourceMetadata;
+
+final class MutatorResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    public function __construct(
+        private readonly ResourceMutatorCollectionInterface $resourceMutators,
+        private readonly OperationMutatorCollectionInterface $operationMutators,
+        private readonly ?ResourceMetadataCollectionFactoryInterface $decorated = null,
+    ) {
+    }
+
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceMetadataCollection = new ResourceMetadataCollection();
+        if ($this->decorated) {
+            $resourceMetadataCollection = $this->decorated->create($resourceClass);
+        }
+
+        $newMetadataCollection = new ResourceMetadataCollection();
+
+        /** @var ResourceMetadata $resource */
+        foreach ($resourceMetadataCollection as $resource) {
+            $resource = $this->mutateResource($resource, $resourceClass);
+            $operations = $this->mutateOperations($resource->getOperations() ?? new Operations());
```

Comment:
While it looks cool in the source code -- it will be calling `mutateOperations` on null values, when it's not necessary.

Personally I would prefer to add checks and don't call `mutateOperations` and `withOperation` when there are no operations.

Though, is it even possible to have resource without operations?

Reply:
In theory, yes we can have a resource without any operations.